From 9132142e1669940b0a88f815e3735ff8db2b17b9 Mon Sep 17 00:00:00 2001
From: Tung Vo <tung.vo.eb@renesas.com>
Date: Sun, 12 Aug 2018 16:09:05 +0700
Subject: [PATCH 511/559] =?UTF-8?q?phy:=20rcar-gen2:=20add=20340=20=CE=BCs?=
 =?UTF-8?q?=20wait=20for=20USB=20PHY=20power=20stable?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

r8a77470: Hardware Manual: Power control and Initialization

The following is the initialize power on procedure of USB subsystem.
	1. Supply 1.8 V (VD181)
	2. Supply 3.3 V (VD331)
	3. HS-USB module stop release and module reset assert.
	4. PLLRESET release
	5. Wait min. 340 μs for USB PHY power stable.
	6. USB0SEL[1:0] should be set to B'10 for USB Function (HS-USB)
	7. LPSTS.SUSPM set to normal mode.
	8. Starting normal operation.

A lack of 340 μs wait after PLLRESET release can lead to some potential
issues. This commit will prevent potential issues.

Signed-off-by: Tung Vo <tung.vo.eb@renesas.com>
---
 drivers/phy/phy-rcar-gen2.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/phy/phy-rcar-gen2.c b/drivers/phy/phy-rcar-gen2.c
index c996473..3c4e09a 100644
--- a/drivers/phy/phy-rcar-gen2.c
+++ b/drivers/phy/phy-rcar-gen2.c
@@ -221,6 +221,8 @@ static int rcar_gen2_phy_power_on(struct phy *p)
 	writel(value, usbhs_base + USBHS_UGCTRL);
 
 	if (of_machine_is_compatible("renesas,r8a77470")) {
+		/* Wait min. 340 μs for USB PHY power stable */
+		udelay(340);
 		/* Power on USB20 PHY */
 		value = readl(usb20_base + USB20_USBCTR_REG);
 		value |= USB20_USBCTR_PLL_RST;
-- 
2.7.4

